zebra-chain/src/block/difficulty.rs

```diff
@@ -181,6 +214,31 @@ impl CompactDifficulty {
             Some(ExpandedDifficulty(result))
         }
     }
+
+    /// Calculate the Work for a compact representation.
+    ///
+    /// See `Definition of Work` in the Zcash Specification, and
+    /// `GetBlockProof()` in zcashd.
+    ///
+    /// Returns None if the corresponding ExpandedDifficulty is None.
+    /// Also returns None on Work overflow, which should be impossible on a
+    /// valid chain.
+    pub fn to_work(&self) -> Option<Work> {
+        let expanded = self.to_expanded();
+
+        if let Some(expanded) = expanded {
```

Comment:
This wont compile because of the block change so you probably dont want to apply this via github but you can use the try operator on Options


```suggestion
        let expanded = self.to_expanded()?;
```